### ISSUE TITLE
Integrate Wazuh with Amazon Security Lake

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#16324]
# Description

This issue proposes to integrate Wazuh with [Amazon Security Lake](https://docs.aws.amazon.com/security-lake/index.html), a data lake for security logs built in the user's account, backed by an S3 bucket and organized as a set of Lake Formation tables. Integrating Wazuh with Amazon Security Lake will allow our users to query and analyze logs from Amazon Security Lake directly within Wazuh, without the need for manual data transfers or additional infrastructure.

# Integration Overview

According to [AWS documentation](https://docs.aws.amazon.com/security-lake/latest/userguide/integrations-third-party.html), the integration may be a `source` integration or a `subscriber` integration. 

This issue will define and track the necessary steps to integrate Wazuh with Amazon Security Lake as both source and subscriber.

## [Integration as a Source](https://github.com/wazuh/wazuh/issues/16362)

Source integrations have the following properties:
- Send data to Security Lake
- Data arrives in Apache Parquet format
- Data arrives in the [Open Cybersecurity Schema Framework (OCSF)](https://docs.aws.amazon.com/security-lake/latest/userguide/open-cybersecurity-schema-framework.html) schema

A summary of the requirements and responsibilities for becoming a custom source for Amazon Security Lake:

 Custom sources are responsible for writing data to the source in accordance with the provided requirements. The registration process includes assigning a unique prefix within the Amazon Security Lake S3 bucket in the user's account, configuring a Glue table and a Glue crawler, and coordinating subscriber access. The data should be written as a set of S3 objects underneath the prefix generated during registration and formatted as Parquet files. The objects must be partitioned by region, AWS account, year, month, day, and hour. The object size should be limited to 1MB (uncompressed), and files written to Amazon Security Lake should buffer 5 minutes of records. Finally, records should be ordered by time within each Parquet-formatted object.

## [Integration as a Subscriber ](https://github.com/wazuh/wazuh/issues/16326)

Subscriber integrations have the following properties:
- Read source data from Security Lake at an HTTPS endpoint or Amazon Simple Queue Service (Amazon SQS) queue, or by directly querying source data from AWS Lake Formation
- Able to read data in Apache Parquet format
- Able to read data in OCSF schema

Becoming a subscriber to Amazon Security Lake involves two access patterns: data access and query access. Subscribers are responsible for handling customer data securely and meeting regulatory requirements. Access to data is given on a per-source basis, and customers can tailor specific access to sources for cost control and least-privilege access control best practices.

In the data access use case, subscribers are notified of new objects for a source as the data is written to the Amazon Security Lake S3 bucket. Access to the S3 bucket and optional SQS object notification queue is done via a role in the user's account.

In the query access use case, Amazon Security Lake employs Lake Formation cross-account table sharing to support direct querying by subscribers of tables residing in the customer’s account. Subscribers need to gather sharing information and accept resource share invitations before creating a resource link table to query the shared resource. Costs incurred by querying data within the subscriber service account fall upon the subscriber.

Subscribers should optimize their queries to take advantage of underlying optimizations, such as controlling partitioning, optimizing file size and format, and enabling partition indexing. To validate correct subscriber functionality, subscribers can create a mock customer account and enable Amazon Security Lake for integration testing.

# Conclusion

In conclusion, integrating Wazuh with Amazon Security Lake as both a source and subscriber will provide Wazuh users with seamless access to security logs stored in Amazon Security Lake. This integration will enhance Wazuh's threat detection and incident response capabilities

**Comment by havidarou** [comment]
Closing this issue. The plan to integrate as a source has been revised and will be implemented in a different way.
